aeron-archive/src/main/java/io/aeron/archive/client/ReplayMerge.java

```diff
@@ -524,6 +533,21 @@ else if (shouldStopAndRemoveReplay(position))
         return workCount;
     }
 
+    private boolean callGetRecordingPosition(final long nowMs, final long correlationId)
+    {
+        if (nowMs >= timeOfNextGetRecordingPositionMs &&
+            archive.archiveProxy().getRecordingPosition(recordingId, correlationId, archive.controlSessionId()))
+        {
+            getRecordingPositionBackoffMs = Long.min(
+                getRecordingPositionBackoffMs * 2, GET_RECORDING_POSITION_BACKOFF_MAX_MS);
+            timeOfNextGetRecordingPositionMs = nowMs + getRecordingPositionBackoffMs;
+
+            return true;
+        }
+
+        return false;
+    }
+
```

Comment:
This method should get a `correlationId` directly if it going to call the `getMaxRecordedPosition` method instead of it being supplied at the call site. Otherwise lots of ids will be wasted during the back-off periods.

Another thing to consider if the back-off should be always applied and not only upon successful calls.

Reply:
Good idea.  I didn't think about the wasted correlationIds.

Reply:
It is more about not doing redundant work than actual waste on the 64-bit value.